src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1782,14 +1787,26 @@ private BaseListSyntax ParseBaseList(SyntaxToken typeKeyword, bool haveParameter
             {
                 _pool.Free(list);
             }
+
+            PostSkipAction skipBadBaseListTokens(ref SyntaxToken colon, SeparatedSyntaxListBuilder<BaseTypeSyntax> list, SyntaxKind expected)
+            {
+                return this.SkipBadSeparatedListTokensWithExpectedKind(ref colon, list,
+                    s_notExpectedForEndOfBaseList,
+                    s_abortForEndOfBaseList,
+                    expected);
+            }
         }
 
-        private PostSkipAction SkipBadBaseListTokens(ref SyntaxToken colon, SeparatedSyntaxListBuilder<BaseTypeSyntax> list, SyntaxKind expected)
+        private static readonly Func<LanguageParser, bool> s_notExpectedForEndOfBaseList = p => p.NotExpectedForEndOfBaseList();
```

Comment:
>s_notExpectedForEndOfBaseList [](start = 59, length = 29)

Why is this necessary? It looks like there is only one use-site and the lambda didn't capture anything, wasn't it cached by the compiler?

Reply:
I filed a follow-up issue for delegate caching after confirming from IL: https://github.com/dotnet/roslyn/issues/46073